### PR TITLE
[SERVER-30428] - Add US-2 and EU-2 as regions

### PR DESCRIPTION
**Pull request by Alaev**
- added new regions
- added option to set region from shared location
- added specs

### CODE REVIEW

File: BringgSDK.js [hunk 1]
```diff
@@ -21,7 +21,16 @@ var BringgSDK = (function () {
...
     "ue1": "https://realtime2-api.bringg.com/",
-    "ew1": "https://eu1-realtime.bringg.com"
+    "ue2": "https://eu2-realtime.bringg.com",
+    "ew1": "https://eu1-realtime.bringg.com",
+    "us2": "https://us2-realtime.bringg.com"
+  };
+
+  var REGIONS_ENUM = {
+    'ew1': 1,
+    'ue1': 2,
```
Comment (roir) [inline BringgSDK.js]: don't you want to change the names to eu instead of ue?

File: BringgSDK.js [hunk 2]
```diff
@@ -673,6 +682,48 @@ var BringgSDK = (function () {
...
+      if (REGIONS_ENUM[regionName] === regionCode) {
+        return regionName;
+      }
+    }
+  };
+
+  /** get region code from shared url */
+  module.getRegionCodeFromUrl = function (url) {
+    var regionCode = url.split('&e=')[1];
+    return parseInt(regionCode);
```
Comment (roir) [inline BringgSDK.js]: parseInt can return a NaN upon failure.
sadly typeof NaN === number apparently so the check you did in the method above won't hold.
so please add an explicit check to either NaN or wrap the parseInt somehow

Comment (Alaev) [reply BringgSDK.js]: I have the check in `module.getRegionByCodeNumber` for parseInt and you can see in the unit test also

```
if (typeof regionCode !== 'number') {
      log('getRegionByCodeNumber: region code must be a number');
      return;
}

```

File: BringgSDK.spec.js [hunk 3]
```diff
@@ -162,8 +164,82 @@ describe('BringgSDK', function () {
...
+
+  describe('getRegionCodeFromUrl', function () {
+    it('should return code from valid shard url', function () {
+      var urlCode = faker.random.number();
+      var url = faker.internet.url() + '&e=' + urlCode;
+      var result = BringgSDK.getRegionCodeFromUrl(url);
+      expect(result).toEqual(urlCode);
     });
 
+    it('should return Nan from not valid url', function () {
```
Comment (Alaev) [inline BringgSDK.spec.js]: NaN unit test

File: BringgSDK.js (hunk 1 above)
Comment (Alaev) [reply BringgSDK.js]: talked about it, some tokens are using this as regions.

**Pull request merged**